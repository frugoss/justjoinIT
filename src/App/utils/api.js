export let API_HOST = 'http://localhost:7000'
if (process.env.NODE_ENV === 'test') {
  API_HOST = 'http://localhost'
}
