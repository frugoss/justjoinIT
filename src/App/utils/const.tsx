

export interface userInterface {
    auth?: boolean,
    name?: string,
    userID?: string,
    loggedPopup?: boolean,
    offPopup?: boolean,
    createPopup?: boolean,
    addPopup?: boolean
}


export interface offerInterface {
    _id: string,
    agreements: string,
    city: string,
    company: string,
    companyIndustry: string,
    companySize: string,
    coordinates: [{ lat: number, lng: number }],
    currency: string,
    description: string,
    email: string,
    employment: string,
    experience: string,
    logo: string | undefined,
    maxSalary: number,
    minSalary: number,
    street: string,
    techStack: [{ language: string, lvl: number }],
    technology: [{ color: string, background: string, img: string, name: string }],
    title: string,
    userID: string,
    website: string,
}

export interface formInterface {
    _id?: string,
    agreements?: string,
    city?: string,
    company?: string,
    companyIndustry?: string,
    companySize?: string,
    coordinates?: [{ lat: number, lng: number }],
    currency?: string,
    description?: string,
    email?: string,
    employment?: string,
    experience?: string,
    logo?: string| undefined,
    maxSalary?: number,
    minSalary?: number,
    street?: string,
    techStack: { language?: string, lvl?: number }[],
    technology?: [{ color: string, background: string, img: string, name: string }],
    title?: string,
    userID: string,
    website?: string,
}


export interface applicationInterface {
    checkbox: boolean,
    email: string,
    file: string,
    message: string,
    name: string,
    offerID: string
}

export interface filtersInterface{
    city: string,
    language: string,
    experience: string,
    salarymax: number,
    salarymin: number,
}

export interface matchInterface{
    city: string,
    language: string,
    experience: string,
    salarymax: string,
    salarymin: string,
}

export interface formValuesCV  {
    name:string,
    email:string,
    message: string,
    checkbox: string,
    file: File,
    offerID: string
}

export interface languageInterface {
    name: string,
    to: string,
    img: string,
    color: string,
    background: string
}



export interface mobileViewModeInterface {
    view: string,
    header: boolean
}

export interface cityInterface {
    name: string,
    to: string,
    coordinates: {lat:number, lng:number}
}
