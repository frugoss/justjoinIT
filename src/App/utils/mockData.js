export const offersList = [
  {
    "_id": "5e9cfa222f9c0e0965b86034",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;baC",
        "color": "rgb(255, 199, 6)",
        "name": "javascript",
        "background": "rgb(255, 199, 6), rgb(255, 175, 0)"
      }
    ],
    "techStack": [
      {
        "language": "JavaScript",
        "lvl": 5
      },
      {
        "language": "ES7",
        "lvl": 3
      }
    ],
    "logo": "data:image/png;basI=",
    "coordinates": [
      {
        "lat": 54.4399555,
        "lng": 18.5440814
      }
    ],
    "city": "Sopot",
    "street": "Tuwima 20",
    "agreements": "Informujemy, że administratorem danych jest dddd z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją.",
    "company": "Travers",
    "companyType": "Software House",
    "companyIndustry": "Medicine",
    "currency": "PLN",
    "employment": "Mandate Contract",
    "experience": "Senior",
    "title": "Javascript developer",
    "minSalary": 15000,
    "maxSalary": 25000,
    "description": "<div><!--block--><strong>Who we are looking for<br></strong><br></div><ul><li><!--block--><strong>Senior Angular Developer (3+ experience)</strong></li><li><!--block-->Good analytical skills (problem solving)</li><li><!--block-->Good communication skills</li><li><!--block-->Enjoy helping other Engineers improve and become better</li><li><!--block-->Very good English speaking skills</li></ul><div><!--block--><strong><br>What we offer<br></strong><br></div><ul><li><!--block-->Working in a Development Team (4 team members) alongside backend team placed in London</li><li><!--block-->Lead a team of talented and collaborative engineers&nbsp;</li><li><!--block-->Help mentor and coach members of the team</li><li><!--block-->Help evolve development standards and practices</li><li><!--block-->Review Pull Requests from other developers</li><li><!--block-->Overall responsible for the technical quality of what the team is producing</li><li><!--block-->Collaborate with back-end developers, UX and UI designers to improve our products and features</li></ul><div><!--block--><strong><br>Benefits<br></strong><br></div><ul><li><!--block-->26 days of paid holidays per year (+ additionally paid bank holidays)</li><li><!--block-->30 days sick leave per year (paid 80%)</li><li><!--block--><strong>Flexible working hours</strong></li><li><!--block-->Work from home (1 day per week)</li><li><!--block--><strong>Taking part in development-related events</strong> (conferences, trainings)</li><li><!--block-->Private medical care</li><li><!--block-->MultiSport Plus Package</li></ul>",
    "companySize": "10 - 40",
    "website": "www.traversy.com"
  },
  {
    "_id": "5e9a411cfe04a70747f1a7d1",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;base64",
        "color": "rgb(31, 94, 182)",
        "name": "python",
        "background": "rgb(31, 94, 182), rgb(31, 123, 196)"
      }
    ],
    "techStack": [
      {
        "language": "Javascript",
        "lvl": 3
      },
      {
        "language": "C#",
        "lvl": 2
      }
    ],
    "logo": "data:image/png;",
    "companyType": "Software House",
    "companySize": "10",
    "companyIndustry": "E-commerce",
    "experience": "Mid",
    "title": "Python Developer",
    "employment": "Permanent",
    "minSalary": 3000,
    "maxSalary": 10000,
    "currency": "EUR",
    "description": "<div><!--block-->Lorem ipsum dolor sit amet, consectetur adipiscing elit. In tincidunt maximus sem vitae cursus. Integer ullamcorper quam quis molestie dictum. Nam eros augue, aliquam vitae tristique in, laoreet ac massa. Suspendisse ultricies vulputate elit, scelerisque pellentesque turpis tempor in. Donec neque est, feugiat sit amet tempor ut, gravida sit amet ligula. Donec elit ante, pharetra ac ex a, malesuada molestie velit. Vivamus cursus risus mi, quis maximus justo convallis vitae.  Nullam ultrices porttitor quam, non cursus risus convallis vel. Mauris feugiat lectus eu arcu tempor, eget hendrerit eros imperdiet. Ut vitae sodales nisi. Nam eu dolor eu libero tincidunt maximus non vitae odio. Cras neque orci, dignissim eu feugiat ac, consectetur ut nisl. Maecenas euismod sem orci. Phasellus eget nulla et sapien vulputate mollis vel tempor sem. Nulla cursus pharetra lacus, lobortis porta nulla eleifend et. Curabitur quis arcu id velit gravida euismod vitae eu turpis. In suscipit, lectus vitae maximus iaculis, mauris massa porttitor est, vel porttitor orci massa ut quam. Morbi in sapien sit amet nisl faucibus posuere.  Quisque sit amet finibus leo. Aenean placerat sit amet nulla rutrum auctor. Curabitur scelerisque eleifend purus, quis bibendum lectus venenatis vitae. Quisque pellentesque lorem ex, eu facilisis magna convallis eget. Aliquam eget fermentum augue. Ut sollicitudin dui elit, eu consectetur mi rutrum in. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam at massa interdum, feugiat augue ac, fringilla mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut velit vulputate nulla luctus aliquet.  Sed ornare viverra felis vitae tincidunt. Pellentesque hendrerit massa sit amet nibh viverra lacinia. Duis vitae metus vel libero tristique euismod. Sed congue consequat purus, quis convallis ex venenatis id. Donec lacus massa, tincidunt vel erat quis, accumsan euismod erat. Curabitur leo dolor, aliquet eget fringilla a, fermentum pulvinar nisl. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.  Proin eleifend sodales ligula in hendrerit. Nam volutpat, enim id varius bibendum, metus orci venenatis nibh, nec vestibulum ex dolor sit amet ipsum. Cras blandit enim vitae justo lobortis, vel efficitur sem luctus. Duis non dolor id erat consectetur eleifend. Nullam ultrices dictum lacus faucibus aliquet. Vestibulum eget diam cursus, fermentum turpis id, aliquet odio. Quisque vitae bibendum nibh, faucibus malesuada mi. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Mauris dictum risus vel metus convallis, vitae cursus metus faucibus. Cras placerat rhoncus enim non tempor. Integer augue odio, efficitur non diam et, luctus lobortis mi. Suspendisse laoreet vestibulum ex non pretium. Nullam rhoncus, risus nec placerat laoreet, dui neque ultrices metus, vitae congue ligula quam a magna. Nullam sodales lacus in aliquet sagittis.</div>",
    "city": "Gdańsk",
    "street": "Tuwima",
    "coordinates": [
      {
        "lat": 54.3659778,
        "lng": 18.6342856
      }
    ],
    "email": "1234@gmail.com",
    "agreements": "Informujemy, że administratorem danych jest _____xDD_________ z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją.",
    "website": "www.lion.com",
    "company": "Lion",
    "remote": true
  },
  {
    "_id": "5e9cf7062f9c0e0965b86029",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;base64",
        "color": "rgb(47, 207, 187)",
        "name": "c",
        "background": "rgb(47, 207, 187), rgb(55, 221, 156)"
      }
    ],
    "techStack": [
      {
        "language": "C",
        "lvl": 4
      },
      {
        "language": "Java",
        "lvl": 2
      },
      {
        "language": "Python",
        "lvl": 5
      }
    ],
    "logo": "data:image/jpeg;base64,",
    "currency": "USD",
    "companyType": "Startup",
    "companyIndustry": "Military",
    "email": "test",
    "employment": "B2B",
    "experience": "Mid",
    "city": "Gdynia",
    "coordinates": [
      {
        "lat": 54.51889,
        "lng": 18.53188
      }
    ],
    "street": "Hubala 10",
    "description": "<div><!--block-->The projects, you will work with, come from the most creative and innovative companies in the Silicon Valley, CA, USA and Western Europe. Take part in the things you have read about on the Internet and do the outstanding stuff here: in Poland. The real adventure is waiting for you.<br><br>About project:<br>Recognition of objects from drone photos, processing into 3D models and providing the user with platform to view and analize the results.<br><br></div><div><!--block-->Expected experience:</div><ul><li><!--block-->Very good spoken and written English</li><li><!--block-->Very good knowledge of Python or Node.JS</li><li><!--block-->Good understanding of Google Cloud, Circle CI and Kuberenetes for CI/CD</li><li><!--block-->Good knowledge of SQL and non-SQL databases</li><li><!--block-->Easy to work with</li><li><!--block-->Ambitious and excited to do new stuff</li></ul><div><!--block--><br>Appreciated knowledge:</div><ul><li><!--block-->Good practices in embedded software development</li><li><!--block-->Good understanding of microservices</li><li><!--block-->Experience in leading a small developers team</li></ul><div><!--block-->&nbsp;</div><div><!--block-->We offer:</div><ul><li><!--block-->International cutting edge projects – work with the technology leaders from CA, USA!</li><li><!--block-->Very flexible working hours, home-office during selected days possible</li><li><!--block-->Corporate-free environment, straightforward, simple, easy</li><li><!--block-->Long-term commitment from apptimia side</li></ul>",
    "title": "C Developer",
    "company": "Fine",
    "minSalary": 3000,
    "maxSalary": 5500,
    "companySize": "20 - 10 ",
    "website": "www.fine.com",
    "agreements": "Informujemy, że administratorem danych jest xD z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją."
  }
]

export const CV = [
  {
    "name": "Sylwester Kowalski",
    "email": "sylwesterkowalski@gmail.com",
    "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet convallis massa. Curabitur efficitur pretium quam ut laoreet. Ut mollis ut velit quis semper. Nam placerat ullamcorper sapien vel laoreet. Vivamus lobortis elit id massa porta condimentum. Curabitur quis nulla eu purus egestas vestibulum. Vivamus semper arcu et blandit faucibus. Fusce sit amet massa cursus, tincidunt leo ut, molestie ligula. Proin sed nisi aliquet, facilisis velit at, scelerisque nunc. Integer sapien leo, porta non pellentesque id, aliquet in magna. Vestibulum venenatis porttitor enim a consequat. Maecenas sed malesuada dolor. Vivamus facilisis laoreet sapien, eu tincidunt dolor bibendum vitae. Etiam id sodales est. Phasellus consequat quam eu nunc tempor auctor.",
    "checkbox": false,
    "offerID": "5e9cfa222f9c0e0965b86034",
    "file": "9fce5a67910bd4d71cb75a400.pdf"
  },
  {
    "name": "Kinga Sylwestrzak",
    "email": "kingasylwestrzak@gmail.com",
    "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet convallis massa. Curabitur efficitur pretium quam ut laoreet. Ut mollis ut velit quis semper. Nam placerat ullamcorper sapien vel laoreet. Vivamus lobortis elit id massa porta condimentum. Curabitur quis nulla eu purus egestas vestibulum. Vivamus semper arcu et blandit faucibus. Fusce sit amet massa cursus, tincidunt leo ut, molestie ligula. Proin sed nisi aliquet, facilisis velit at, scelerisque nunc. Integer sapien leo, porta non pellentesque id, aliquet in magna. Vestibulum venenatis porttitor enim a consequat. Maecenas sed malesuada dolor. Vivamus facilisis laoreet sapien, eu tincidunt dolor bibendum vitae. Etiam id sodales est. Phasellus consequat quam eu nunc tempor auctor.",
    "checkbox": true,
    "offerID": "5e9cfa222f9c0e0965b86034",
    "file": "9fce5a67910bd4d71cb75a401.pdf"
  }
]
