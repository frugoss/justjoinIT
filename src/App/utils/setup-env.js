import {server} from 'App/utils/server.js'
beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())
