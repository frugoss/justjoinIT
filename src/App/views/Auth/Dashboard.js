import React, {useEffect, useState} from 'react'
import OffersList from "../Home/OffersList";
import styles from './auth.module.scss'

const Dashboard = ({user,offers}) => {
    const [applications, setApplications] = useState()
    const offersList = offers.filter(offer => offer.userID === user.userID)
    useEffect( () => {
            fetch('http://192.168.10.25:7000/cv/get',{method: 'GET'}).then(function (response) {
                response.json().then(json => {setApplications(json)
                })

            })
        },[])
    return(
    <div className={styles.dashboard}>
    <OffersList applications={applications} offersList={offersList}/>
    </div>
    )
}

export default Dashboard
