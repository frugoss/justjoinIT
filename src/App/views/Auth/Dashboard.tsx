import React, {useEffect, useState} from 'react'
import OffersList from "../Home/OffersList";
import styles from './auth.module.scss'
import {userInterface} from "../../utils/const";
import {offerInterface} from "../../utils/const";
import {API_HOST} from "../../utils/api";
type DashboardProps = {
    user: userInterface,
    offers: offerInterface[]
}
const Dashboard: React.FC<DashboardProps> = ({user,offers}) => {
    const [applications, setApplications] = useState([])
    const offersList = offers.filter(offer => offer.userID === user.userID)
    useEffect( () => {
            fetch(`${API_HOST}/cv/get`,{method: 'GET'}).then(function (response) {
                response.json().then(json => {setApplications(json)
                })

            })
        },[])

    return(
    <div className={styles.dashboard}>
        <OffersList applications={applications} offersList={offersList} />
    </div>
    )
}

export default Dashboard
