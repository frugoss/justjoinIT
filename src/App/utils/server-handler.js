import {rest} from 'msw'
import {offersList, CV} from 'App/utils/mockData.js'

const handlers = [
  rest.post('/devs', async (req, res, ctx) => {
    return res(ctx.text("4325325325235"))
  }),
  rest.post('/devs/signup', async (req, res, ctx) => {
    return res(ctx.text("4325325325235"))
  }),
  rest.get('/', async (req, res, ctx) => {
    return res(ctx.json(offersList))
  }),
  rest.get('/cv/get', async (req, res, ctx) => {
    return res(ctx.json(CV))
  })
]

export {handlers}
