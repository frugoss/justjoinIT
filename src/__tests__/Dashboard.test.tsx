import Dashboard from "../App/views/Auth/Dashboard";
import React from 'react'
import {screen,waitFor} from "@testing-library/react";
import {offersList} from '../App/utils/mockData';
import {renderWithRouter} from "../App/utils/tests";

test('Shows offers posted by a company', async () => {
    const {container} = renderWithRouter(<Dashboard user={{userID: "5e8cc6d853004b069507091d"}} offers={offersList}/>)
    await waitFor( () => expect(screen.getAllByRole("heading").length).toEqual(3))
    expect(container).toMatchSnapshot();
})

test("Shows a prompt to add an offer if there are no offers posted", async () => {
    renderWithRouter(<Dashboard user={{userID: "483ds90f49f890d8f5489"}} offers={offersList}/>, {route: "/dashboard"})
    await waitFor( () => expect(screen.getByText(/to see it on your dashboard/i)).toBeInTheDocument())
})


test('Checks the number of offers with 2 cvs', async () => {

    renderWithRouter(<Dashboard user={{userID: "5e8cc6d853004b069507091d"}} offers={offersList}/>)
    await waitFor(() => expect(screen.getAllByText("2 CV")).toHaveLength(1))
})


test('Shows the offers which dont have CVs', async () => {
    renderWithRouter(<Dashboard user={{userID: "5e8cc6d853004b069507091d"}} offers={offersList}/>)
    await waitFor(() => expect(screen.getAllByText("0 CV")).toHaveLength(2))
})
