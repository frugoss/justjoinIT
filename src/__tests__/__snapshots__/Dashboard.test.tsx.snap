// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows offers posted by a company 1`] = `
<div>
  <div
    class="dashboard"
  >
    <div
      class="offersList"
    >
      
      <a
        class="offerLink"
        href="/offers/travers-javascript-developer"
      >
        <div
          class="main animated fadeIn fast"
        >
          <div
            class="color"
            style="background: rgb(255, 199, 6);"
          />
          <div
            class="img"
          >
            <img
              alt="logo"
              class="logoList"
              src="data:image/png;basI="
            />
          </div>
          <div
            class="card"
          >
            <div
              class="flex fontSize"
            >
              <div>
                <h3>
                  Javascript developer
                </h3>
              </div>
              <div
                class="salary"
              >
                <span>
                  15 000
                </span>
                 - 
                <span>
                  25 000
                </span>
                 
                PLN
              </div>
            </div>
            <div
              class="flex fontIconText"
            >
              <div
                class="businessContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
                  />
                </svg>
                 
                Travers
                 
              </div>
              <div
                class="locationContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-margin-1 MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 4.17 4.42 9.92 6.24 12.11.4.48 1.13.48 1.53 0C14.58 18.92 19 13.17 19 9c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                  />
                </svg>
                Sopot
              </div>
            </div>
          </div>
          <div
            class="iconCv"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root iconCvSize"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
              />
            </svg>
            <div
              class="dashboardCVContainer"
            >
              0
               CV
            </div>
          </div>
        </div>
      </a>
      <a
        class="offerLink"
        href="/offers/lion-python-developer"
      >
        <div
          class="main animated fadeIn fast"
        >
          <div
            class="color"
            style="background: rgb(31, 94, 182);"
          />
          <div
            class="img"
          >
            <img
              alt="logo"
              class="logoList"
              src="data:image/png;"
            />
          </div>
          <div
            class="card"
          >
            <div
              class="flex fontSize"
            >
              <div>
                <h3>
                  Python Developer
                </h3>
              </div>
              <div
                class="salary"
              >
                <span>
                  3 000
                </span>
                 - 
                <span>
                  10 000
                </span>
                 
                EUR
              </div>
            </div>
            <div
              class="flex fontIconText"
            >
              <div
                class="businessContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
                  />
                </svg>
                 
                Lion
                 
              </div>
              <div
                class="locationContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-margin-1 MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 4.17 4.42 9.92 6.24 12.11.4.48 1.13.48 1.53 0C14.58 18.92 19 13.17 19 9c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                  />
                </svg>
                Gda≈Ñsk
              </div>
            </div>
          </div>
          <div
            class="iconCv"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root iconCvSize"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
              />
            </svg>
            <div
              class="dashboardCVContainer"
            >
              0
               CV
            </div>
          </div>
        </div>
      </a>
      <a
        class="offerLink"
        href="/offers/fine-c-developer"
      >
        <div
          class="main animated fadeIn fast"
        >
          <div
            class="color"
            style="background: rgb(47, 207, 187);"
          />
          <div
            class="img"
          >
            <img
              alt="logo"
              class="logoList"
              src="data:image/jpeg;base64,"
            />
          </div>
          <div
            class="card"
          >
            <div
              class="flex fontSize"
            >
              <div>
                <h3>
                  C Developer
                </h3>
              </div>
              <div
                class="salary"
              >
                <span>
                  3 000
                </span>
                 - 
                <span>
                  5 500
                </span>
                 
                USD
              </div>
            </div>
            <div
              class="flex fontIconText"
            >
              <div
                class="businessContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
                  />
                </svg>
                 
                Fine
                 
              </div>
              <div
                class="locationContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-margin-1 MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 4.17 4.42 9.92 6.24 12.11.4.48 1.13.48 1.53 0C14.58 18.92 19 13.17 19 9c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                  />
                </svg>
                Gdynia
              </div>
            </div>
          </div>
          <div
            class="iconCv"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root iconCvSize"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
              />
            </svg>
            <div
              class="dashboardCVContainer"
            >
              0
               CV
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
